qiskit_nature/operators/second_quantization/star_algebra.py

```diff
@@ -0,0 +1,113 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sum Operator base interface."""
+
+from abc import ABC, abstractmethod
+
+from .multiply import MultiplyMixin
+# TODO: Move import path after Terra#5617 is merged.
+
+
+class StarAlgebraMixin(MultiplyMixin, ABC):
+    """The star algebra mixin class.
+
+    Star algebra is an algebra with adjoint (dagger).
+
+    This class overrides:
+
+        - ``*``, ``__mul__`` -> :meth:`mul`
+        - ``+``, ``__add__`` -> :meth:`add`
+        - ``@``, ``__matmul__`` -> :meth:`compose`
+
+    The following abstract methods must be implemented by subclasses:
+
+        - :meth:``mul(self, other)``
+        - :meth:``add(self, other)``
+        - :meth:``compose(self, other)``
+    """
+
+    # Scalar multiplication
+
+    @abstractmethod
+    def mul(self, other: complex):
+        r""" Return scalar multiplication of self and other, overloaded by ``*``."""
+        return NotImplementedError
+
+    def __mul__(self, other: complex):
+        return self.mul(other)
+
+    def _multiply(self, other: complex):
+        return self.mul(other)
+
+    # Addition, substitution
+
+    @abstractmethod
+    def add(self, other):
+        r""" Return Operator addition of self and other, overloaded by ``+``."""
+        return NotImplementedError
+
+    def __add__(self, other):
+        return self.add(other)
+
+    def __radd__(self, other):
+        if other == 0:
+            return self
+        return self.add(other)
+
+    def __sub__(self, other):
+        return self.add(-other)
+
+    def _add(self, other):
+        return self.add(other)
```

Comment:
I guess I was expecting `__add__` to call `self._add()` where `_add()` was the abstract method that would be implemented by a subclass. But there is `add()` which seems to have this role and its public so I guess you can use an operator or call .add(). I have not figured the role for _add()

Reply:
Thanks. Sorry, I forgot to remove this private method. (I have added this when I try to use LinearMixin but now I decided the implementation is future work.) I will remove it.